tests/integration/integrations/deepchecks/materializers/test_deepchecks_result_materializer.py

```diff
@@ -13,47 +13,53 @@
 #  permissions and limitations under the License.
 """Unit tests for Deepchecks result materializer."""
 
+import sys
 from contextlib import ExitStack as does_not_raise
 
 import pandas as pd
 import pytest
-from deepchecks.core.check_result import CheckResult
-from deepchecks.core.suite import SuiteResult
-from deepchecks.tabular import Context, Dataset, TrainTestCheck
 
 from tests.unit.test_general import _test_materializer
-from zenml.integrations.deepchecks.materializers.deepchecks_results_materializer import (
-    DeepchecksResultMaterializer,
-)
 
 
-class DatasetSizeComparison(TrainTestCheck):
-    """Check which compares the sizes of train and test datasets."""
+@pytest.fixture
+def check_result():
+    """Fixture to get a check result."""
+    from deepchecks.core.check_result import CheckResult
+    from deepchecks.tabular import Context, Dataset, TrainTestCheck
+
+    class DatasetSizeComparison(TrainTestCheck):
+        """Check which compares the sizes of train and test datasets."""
 
-    def run_logic(self, context: Context) -> CheckResult:
-        ## Check logic
-        train_size = context.train.n_samples
-        test_size = context.test.n_samples
+        def run_logic(self, context: Context) -> CheckResult:
+            ## Check logic
 
-        ## Return value as check result
-        return_value = {"train_size": train_size, "test_size": test_size}
-        return CheckResult(return_value)
+            train_size = context.train.n_samples
+            test_size = context.test.n_samples
 
+            ## Return value as check result
+            return_value = {"train_size": train_size, "test_size": test_size}
+            return CheckResult(return_value)
 
-@pytest.fixture
-def check_result():
-    """Fixture to get a check result."""
     train_dataset = Dataset(
         pd.DataFrame(data={"x": [1, 2, 3, 4, 5, 6, 7, 8, 9]}), label=None
     )
     test_dataset = Dataset(pd.DataFrame(data={"x": [1, 2, 3]}), label=None)
     return DatasetSizeComparison().run(train_dataset, test_dataset)
 
 
+@pytest.mark.skipif(
+    sys.version_info.minor == 12,
+    reason="The deepchecks integrations is not yet supported on 3.12.",
+)
```

Comment:
Why did you have to upgrade deepchecks if it's not even supported on 3.12 ?

Reply:
It is due to the `pandas` upgrade. The older version of `deepchecks` that we were using (regardless of the python version) had an uncapped `pandas` dependency. Because of this, even when you do `pip install deepchecks`, it ends up installing a version of `pandas` that is not compatible with this older version of `deepchecks`.